readthedocs/rtd_tests/tests/test_repo_parsing.py

```diff
@@ -22,26 +22,61 @@ def test_github(self):
+
     def test_gitlab(self):
         self.pip.repo = 'https://gitlab.com/user/repo'
         self.assertEqual(self.version.get_gitlab_url(docroot='/foo/bar/', filename='file'), 'https://gitlab.com/user/repo/blob/master/foo/bar/file.rst')
 
         self.pip.repo = 'https://gitlab.com/user/repo/'
         self.assertEqual(self.version.get_gitlab_url(docroot='/foo/bar/', filename='file'), 'https://gitlab.com/user/repo/blob/master/foo/bar/file.rst')
 
+        self.pip.repo = 'https://gitlab.com/user/repo.gitlab.io'
+        self.assertEqual(self.version.get_gitlab_url(docroot='/foo/bar/', filename='file'), 'https://gitlab.com/user/repo.gitlab.io/blob/master/foo/bar/file.rst')
+
+        self.pip.repo = 'https://gitlab.com/user/repo.gitlab.io/'
+        self.assertEqual(self.version.get_gitlab_url(docroot='/foo/bar/', filename='file'), 'https://gitlab.com/user/repo.gitlab.io/blob/master/foo/bar/file.rst')
+
         self.pip.repo = 'https://gitlab.com/user/repo.git'
         self.assertEqual(self.version.get_gitlab_url(docroot='/foo/bar/', filename='file'), 'https://gitlab.com/user/repo/blob/master/foo/bar/file.rst')
 
+        self.pip.repo = 'https://gitlab.com/user/repo.gitlab.io.git'
+        self.assertEqual(self.version.get_gitlab_url(docroot='/foo/bar/', filename='file'), 'https://gitlab.com/user/repo.gitlab.io/blob/master/foo/bar/file.rst')
+
+        self.pip.repo = 'https://gitlab.com/user/repo.git.git'
+        self.assertEqual(self.version.get_gitlab_url(docroot='/foo/bar/', filename='file'), 'https://gitlab.com/user/repo.git/blob/master/foo/bar/file.rst')
+
     def test_bitbucket(self):
         self.pip.repo = 'https://bitbucket.org/user/repo'
         self.assertEqual(self.version.get_bitbucket_url(docroot='/foo/bar/', filename='file'), 'https://bitbucket.org/user/repo/src/master/foo/bar/file.rst')
 
         self.pip.repo = 'https://bitbucket.org/user/repo/'
         self.assertEqual(self.version.get_bitbucket_url(docroot='/foo/bar/', filename='file'), 'https://bitbucket.org/user/repo/src/master/foo/bar/file.rst')
 
+        self.pip.repo = 'https://bitbucket.org/user/repo.gitbucket.io'
+        self.assertEqual(self.version.get_bitbucket_url(docroot='/foo/bar/', filename='file'), 'https://bitbucket.org/user/repo.gitbucket.io/src/master/foo/bar/file.rst')
+
+        self.pip.repo = 'https://bitbucket.org/user/repo.gitbucket.io/'
+        self.assertEqual(self.version.get_bitbucket_url(docroot='/foo/bar/', filename='file'), 'https://bitbucket.org/user/repo.gitbucket.io/src/master/foo/bar/file.rst')
+
         self.pip.repo = 'https://bitbucket.org/user/repo.git'
-        self.assertEqual(self.version.get_bitbucket_url(docroot='/foo/bar/', filename='file'), 'https://bitbucket.org/user/repo/src/master/foo/bar/file.rst')
+        self.assertEqual(self.version.get_bitbucket_url(docroot='/foo/bar/', filename='file'), 'https://bitbucket.org/user/repo.git/src/master/foo/bar/file.rst')
```

Comment:
I don't know how to handle this.. Both clone urls are valid for a project with name `name.git`, yo can do a clone like `git clone https://bitbucket.org/user/name.git` or `git clone https://bitbucket.org/user/name.git.git`

Reply:
But on the web, you only can access to the root path like `https://bitbucket.org/user/name.git/src/master/foo/bar/file.rst`

Reply:
And, bitbutcked gives a url in the form of `https://user@bitbucket.org/user/name.git.git` for https clone, so I guess this probably never happen (but we need another regex for this case)